pkgs/by-name/po/posting/package.nix

```diff
@@ -0,0 +1,156 @@
+  httpx = (
+    py-pkgs.httpx.overrideAttrs (oldAttrs: rec {
+      version = "0.27.2";
+      src = fetchFromGitHub {
+        owner = "encode";
+        repo = oldAttrs.pname;
+        rev = "refs/tags/${version}";
+        hash = "sha256-N0ztVA/KMui9kKIovmOfNTwwrdvSimmNkSvvC+3gpck=";
+      };
+      nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ py-pkgs.zstandard ];
+    })
+  );
+
+  # Current nixpkgs version is too outdated
+  textual = (
+    py-pkgs.textual.overrideAttrs (oldAttrs: rec {
+      version = "0.85.0";
+      src = fetchFromGitHub {
+        owner = "Textualize";
+        repo = "textual";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-ROq/Pjq6XRgi9iqMlCzpLmgzJzLl21MI7148cOxHS3o=";
+      };
+    })
+  );
+
+  # Current nixpkgs version is too outdated
+  pydantic = (
+    py-pkgs.pydantic.overrideAttrs (oldAttrs: rec {
+      version = "2.9.2";
+      src = fetchFromGitHub {
+        owner = "pydantic";
+        repo = "pydantic";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-Eb/9k9bNizRyGhjbW/LAE/2R0Ino4DIRDy5ZrQuzJ7o=";
+      };
+      propagatedBuildInputs = [
+        pydantic-core
+        py-pkgs.annotated-types
+        py-pkgs.jsonschema
+      ];
+    })
+  );
+
+  # Exact version of core is required by pydantic
+  pydantic-core = (
+    py-pkgs.pydantic-core.overrideAttrs (oldAttrs: rec {
+      version = "2.23.4";
+      src = fetchFromGitHub {
+        owner = "pydantic";
+        repo = "pydantic-core";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-WSSwiqmdQN4zB7fqaniHyh4SHmrGeDHdCGpiSJZT7Mg=";
+      };
+
+      cargoDeps = rustPlatform.fetchCargoTarball {
+        inherit src;
+        name = "${oldAttrs.pname}-${version}";
+        hash = "sha256-dX3wDnKQLmC+FabC0van3czkQLRcrBbtp9b90PgepZs=";
+      };
+
+      nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ py-pkgs.typing-extensions ];
+    })
+  );
+
+  # Not available in nixpkgs
+  textual-autocomplete = py-pkgs.buildPythonPackage rec {
+    pname = "textual_autocomplete";
+    version = "3.0.0a12";
+    pyproject = true;
+    # Alpha version of this package are only available on Pypi.
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-HSyeTSTH9XWryMYSy2q//0cG9qqrm5OVBrldroRUkwk=";
+    };
+
+    nativeBuildInputs = [ py-pkgs.poetry-core ];
+
+    dependencies = [
+      textual
+      py-pkgs.typing-extensions
+    ];
+  };
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "posting";
+  version = "2.0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "darrenburns";
+    repo = "posting";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-6KtC5VuG3x07VTenpyDAJr9KO4jdTCFk1u/pSoyYPsc=";
+  };
+
+  # We patch the pyproject.toml to allow a slightly outdated watchfiles dependency.
+  # The correct version is not in nixpkgs yet and overriding the version is not trivial.
+  # (i tried)
+  patches = [ ./allow-watchfiles-v22.patch ];
```

Comment:
Most requirements could probably be patched to the current nixpkgs version. Is this preferred over making sure the standard pyproject.toml is satisfied?

Reply:
Its case by case to be completely honest. I think just due to the sheer amount of overriding youre doing (which is a maintenance hassle) itd be generally easier for you and others if it was compatible with nixpkgs as-is. Of course, if its not possible thats a different problem.

Reply:
From matrix: https://matrix.to/#/!kjdutkOsheZdjqYmqp%3Anixos.org/%241sPNgtk0FyKTPgCoj5JHYfRZe4ZJ5Fq6d-zsNhRKoYM

Since this is an app it should be fine to use overrides here.

Reply:
Alright I've made the pyproject use the nixpkgs version for everything except pydantic-settings and textual. The former causes a crash when I use the nixpkgs version and the latter is explicitly pinned by the maintainer of posting so I assume it will cause breakages if I relax that.